xio-core/src/main/java/com/xjeffrose/xio/http/ClientPool.java

```diff
@@ -65,6 +66,7 @@ public Client acquire(ClientConfig config, Supplier<Client> clientSupplier) {
               if (available) {
                 int count = meta.usageCount.incrementAndGet();
                 log.debug("reusing client in pool with usage count {}", count);
+                meta.client.prepareForReuse(() -> new ProxyBackendHandler(ctx));
```

Comment:
here is the fix to the second issue - replacing the "app handler" associated with the appropriate "front end"